src/lang/modifyAst/boolean.ts

```diff
@@ -0,0 +1,346 @@
+}
+
+export async function applyUnionFromTargetOperatorSelections(
+  solids: Selections,
+  dependencies: {
+    kclManager: KclManager
+    engineCommandManager: EngineCommandManager
+    codeManager: CodeManager
+  }
+): Promise<Error | void> {
+  const ast = dependencies.kclManager.ast
+
+  const artifacts: Artifact[] = []
+  for (const selection of solids.graphSelections) {
+    if (selection.artifact) {
+      artifacts.push(selection.artifact)
+    }
+  }
+
+  if (artifacts.length < 2) {
+    return new Error('Not enough artifacts selected')
+  }
+
+  const orderedChildrenEach = artifacts.map((artifact) =>
+    findAllChildrenAndOrderByPlaceInCode(
+      artifact,
+      dependencies.engineCommandManager.artifactGraph
+    )
+  )
+
+  const lastVars: VariableDeclaration[] = []
+  for (const orderedArtifactLeafs of orderedChildrenEach) {
+    const lastVar = getLastVariable(orderedArtifactLeafs, ast)
+    if (!lastVar) continue
+    lastVars.push(lastVar.variableDeclaration.node)
+  }
+
+  if (lastVars.length < 2) {
+    return new Error('Not enough variables found')
+  }
+
+  const modifiedAst = booleanUnionAstMod({
+    ast,
+    solids: lastVars,
+  })
+  const updateAstResult = await dependencies.kclManager.updateAst(
+    modifiedAst,
+    false
+  )
+  await dependencies.codeManager.updateEditorWithAstAndWriteToFile(
+    updateAstResult.newAst
+  )
+  await dependencies.kclManager.updateAst(modifiedAst, true)
+}
+export async function applyIntersectFromTargetOperatorSelections(
+  solids: Selections,
+  dependencies: {
+    kclManager: KclManager
+    engineCommandManager: EngineCommandManager
+    codeManager: CodeManager
+  }
+): Promise<Error | void> {
+  const ast = dependencies.kclManager.ast
+
+  const artifacts: Artifact[] = []
+  for (const selection of solids.graphSelections) {
+    if (selection.artifact) {
+      artifacts.push(selection.artifact)
+    }
+  }
+
+  if (artifacts.length < 2) {
+    return new Error('Not enough artifacts selected')
+  }
+
+  const orderedChildrenEach = artifacts.map((artifact) =>
+    findAllChildrenAndOrderByPlaceInCode(
+      artifact,
+      dependencies.engineCommandManager.artifactGraph
+    )
+  )
+
+  const lastVars: VariableDeclaration[] = []
+  for (const orderedArtifactLeafs of orderedChildrenEach) {
+    const lastVar = getLastVariable(orderedArtifactLeafs, ast)
+    if (!lastVar) continue
+    lastVars.push(lastVar.variableDeclaration.node)
+  }
+
+  if (lastVars.length < 2) {
+    return new Error('Not enough variables found')
+  }
+
+  const modifiedAst = booleanIntersectAstMod({
+    ast,
+    solids: lastVars,
+  })
+  const updateAstResult = await dependencies.kclManager.updateAst(
+    modifiedAst,
+    false
+  )
+  await dependencies.codeManager.updateEditorWithAstAndWriteToFile(
+    updateAstResult.newAst
+  )
+  await dependencies.kclManager.updateAst(modifiedAst, true)
+}
+
+export function findAllChildrenAndOrderByPlaceInCode(
```

Comment:
@Irev-Dev when you have a chance could you add a little description here, especially what we mean by children? Seems like we could reuse this elsewhere, and I think I understand most of it but would be good to have IMO! Not of a blocker ofc

Reply:
Yeah I added

> returns all children of a given artifact, and sorts them DESC by start sourceRange 
The usecase is we want the last declare relevant  child to use in the boolean operations
but might be useful else where